WordPress/gutenberg
# Add an interactive variant to `@wordpress/create-block`

## What?



Add an `interactive` variant to the `@wordpress/create-block` package.

## Why?



So people can use this variant to start testing the Interactivity API.

## How?



I followed what @ryanwelcher did in [this PR](https://github.com/WordPress/gutenberg/pull/41289) and added a new variant for an `interactive` block.

I also added the possibility of adding a `viewScript` field to the `block.json` file.

## Testing Instructions






- Run the `@wordpress/create-block` package using:
  - `cd packages/create-block`
  - `node index.js`
- Choose the `interactive` variant.
- Fill in the rest of the answers.
- Take a look at the files:
  - The `block.json` should have `supports.interactivity: true` and a `viewScript` field.
  - The `render.php` file should contain directives.
  - There should be a `view.js` file with an import of `@wordpress/interactivity` and a call to `store()`.

If you want to try that the interactivity works, you can use `wp-now`:

- `cd (installation directory) && npm start`
- `npx @wp-now/wp-now start` (in another terminal)
- Go to wp-admin
- Install Gutenberg 16.2 or higher
- Add a new post or edit an existing one
- Add the new block
- Save the post
- Go to the frontend
- Check that the toggle works

## Screenshots or screencast 

<a href="https://www.loom.com/share/3680f3b779e14306b8a7fd5bb484988c">
    <img style="max-width:300px;" src="https://cdn.loom.com/sessions/thumbnails/3680f3b779e14306b8a7fd5bb484988c-with-play.gif">
  </a>

https://www.loom.com/share/3680f3b779e14306b8a7fd5bb484988c

## Questions

- Do we want to install `wp-now` and add npm scripts to run it?
- If so, only for the interactive or for all the examples?
- Is it enough with this basic example?
